pkg/auth/handler/forgotpwd/reset_form.go

```diff
@@ -114,11 +115,11 @@ func (h ForgotPasswordResetFormHandler) prepareResultTemplateContext(r *http.Req
 	var payload ForgotPasswordResetFormPayload
 	payload, err = decodeForgotPasswordResetFormRequest(r)
 	if err != nil {
-		err = skyerr.NewBadRequest("invalid form data")
+		err = skyerr.NewBadRequest("invalid request form")
 		return
 	}
 
-	if err = payload.Validate(); err != nil {
+	if err = h.Validator.WithMessage("invalid request form").ValidateGoValue("#ForgotPasswordResetForm", payload); err != nil {
```

Comment:
Is it possible to remove ValidateGoValue? I know we are validating a url.Values here. Can we first encode it to JSON to preserve the original form and then use ValidateReader? It allows us to specify `additionalProperties: false` to catch typo. We should specify `additionalProperties: false` whenever possible.

Reply:
I don't think using `additionalProperties: false` would be forward compatible: a newer client that add extra fields in request would not be able to call old version of API. Do you think we should have this level of compatibility?

Reply:
This issue is related to explicit API versioning. Let's not to include additionalProperties for now.